opensearchpy/connection/http_async.py

```diff
@@ -190,17 +191,20 @@ async def perform_request(
             body = self._gzip_compress(body)
             req_headers["content-encoding"] = "gzip"
 
-        req_headers = {
-            **req_headers,
-            **self._http_auth(method, url, query_string, body),
-        }
+        auth = self._http_auth if isinstance(self._http_auth, aiohttp.BasicAuth) else None
+        if callable(self._http_auth):
+            req_headers = {
+                **req_headers,
+                **self._http_auth(method, url, query_string, body),
+            }
```

Comment:
I believe `auth` is already `None` (if not it should be), isn't it, and these are mutually exclusive?

```
if isinstance(self._http_auth, aiohttp.BasicAuth):
   auth = self._http_auth
elif callable(self._http_auth):
   req_headers = {
     **req_headers,
     **self._http_auth(method, url, query_string, body),
   }
```

Not a must have, just thinking about making it more readable.

Reply:
`auth` is used in `self.session.request` a bit further down. Doing it this way will, when a callable is used for auth, raise a `NameError` because `auth` isn't defined. `auth=None` is the default to `session.request()` yes, but the only other alternative is to make 2 branches to call `self.session.request` with the different iterations, which I think is a lot messier. Doing it this way allows a single call to `session.request(auth=auth)`, where `auth` is `None` (the default), or an instance of `aiohttp.BasicAuth`.